lib/kernel/src/application_controller.erl

```diff
@@ -1799,48 +1805,99 @@ do_config_diff([{Env, Value} | AppEnvNow], AppEnvBefore, {Changed, New}) ->
     end.
 
 
+conf_param_to_conf({config, FileName}) ->
+    BFName = filename:basename(FileName,".config"),
+    FName = filename:join(filename:dirname(FileName),
+                          BFName ++ ".config"),
+    case load_file(FName) of
+        {ok, NewEnv} ->
+            %% OTP-4867 sys.config may now contain names of other
+            %% .config files as well as configuration parameters.
+            %% Therefore read and merge contents.
+            if
+                BFName =:= "sys" ->
+                    DName = filename:dirname(FName),
+                    {ok, SysEnv, Errors} =
```

Comment:
So including additional config files in a config is still restricted to only a file named `sys.config` and thus looks like its not supported in the config passed with `configfd`?

Reply:
The problem is that we do not know where to look for the extra files when using `-configfd`. For `sys.config` it looks in the same directory as `sys.config`. I suppose if we could do it if we only allow absolute file names, or if we make the paths relative to the emulators cwd.

Reply:
Ah, didn't think about that. If it was done I would go for relative to the root of the current version of the release. So a config file in `releases/VSN/` can be referenced with `./`.

Reply:
We will add so that it will look for other files in `releases/VSN` if running in a release. Is there anywhere else that this could be used within relx/rebar3 so that more directories need to be added?

Reply:
Not in what we natively support, but I could expect someone to use /tmp mounted to a ramfs or tmpfs partition to get writeable configs going in some embedded environments where they want to generate the script without doing it with fds. 

I would possibly assume something like:

1. start with the cwd of the release
2. look in the directory of the first config file passed

To be fair though, this is quickly becoming hella obtuse and I don't think it would be mandatory to support all of that as of today since I'm not sure we even do right now and people can still use absolute filenames for that.

Reply:
> To be fair though, this is quickly becoming hella obtuse and I don't think it would be mandatory to support all of that as of today since I'm not sure we even do right now and people can still use absolute filenames for that.

Hmm, yes that is true. Maybe it is enough to only support absolute paths?

For releases I suppose they could use `$PWD/release/$VSN/` in the sys.config.src for any files that they need to include.

Reply:
I'd make a non-absolute path relative to the release's ROOTDIR (or whatever it is we do today), and absolute has no problem regardless. That way you support both instantly.

Reply:
Will environment variables work when used in a config file?

I would suggest doing the releases dir and absolute, I think that is enough and not going to be confusing.

Reply:
> Will environment variables work when used in a config file?

They will work as before, i.e. not at all. I was talking about the way relx helps with `extended_bin`.

> I would suggest doing the releases dir and absolute, I think that is enough and not going to be confusing.

Agreed.

Reply:
So, when looking at implementing looking for config files in release dir we realized that we do not know what the release dir is when starting erlang. The release dir can be set using a sasl configuration parameter, which is not available during boot.

So I think that for now, we will have to make do with absolute paths and if really needed we need to add some command-line option to set the cwd for configfd config.

Reply:
Does it know the location of the boot script being run?

Reply:
Yes it does, that may work as an indicator...

Reply:
Yea, in the case of relx generated releases that will be the same as the release dir `releases/VSN/`. Obviously it isn't guaranteed to be the same but if its documented as "cwd is the directory of the boot script being booted" instead of "the release directory" I'd think it would be fine.

Reply:
One thing to note is that release_handler assumes a very specific path, and if you change it (unless the SASL thing you mentioned allows changes), relups start breaking. See https://bugs.erlang.org/browse/ERL-715

For the time being, assuming any other directory for configs sounds dangerous because it just breaks OTP functionality to change it, and as far as I know this isn't documented anywhere but has been in forever.

Reply:
Yes, release upgrades will not work with `configfd` style config for the new version. I was thinking that maybe it would be a good idea to allow release handler to take the contents of a config file as an argument when doing an upgrade that does not involve en emulator restart. When doing an emulator restart, the heart command will have to deal with supplying the correct `-configfd` setup.

Reply:
I have added support for including files. If a file path is relative, the path will first be searched relative to the directory where the boot script is located and then relative to the current working directory of the emulator.